tests/core/engine_adapter/test_snowflake.py

```diff
@@ -449,3 +449,198 @@ def test_materialized_view_properties(make_mocked_engine_adapter: t.Callable):
     assert sql_calls == [
         'CREATE OR REPLACE MATERIALIZED VIEW "test_table" CLUSTER BY ("a") AS SELECT 1',
     ]
+
+
+def test_creatable_type_secure_view(make_mocked_engine_adapter: t.Callable):
+    adapter = make_mocked_engine_adapter(SnowflakeEngineAdapter)
+
+    adapter.create_view(
+        "test_table",
+        parse_one("SELECT 1"),
+        view_properties={
+            "creatable_type": exp.Column(this=exp.Identifier(this="secure")),
+        },
+    )
+
+    sql_calls = to_sql_calls(adapter)
+    # https://docs.snowflake.com/en/sql-reference/sql/create-view.html
+    assert sql_calls == [
+        'CREATE OR REPLACE SECURE VIEW "test_table" AS SELECT 1',
+    ]
+
+
+def test_creatable_type_secure_materialized_view(make_mocked_engine_adapter: t.Callable):
+    adapter = make_mocked_engine_adapter(SnowflakeEngineAdapter)
+
+    adapter.create_view(
+        "test_table",
+        parse_one("SELECT 1"),
+        secure=True,
+        materialized=True,
+        view_properties={
+            "creatable_type": exp.Column(this=exp.Identifier(this="secure")),
```

Comment:
Why do we set both `view_properties` and `secure=True` here? Is the latter correct?

Reply:
That is legacy from my first attempt 😛